### DISCUSSION TITLE
How can I retrain a model after checking Concept Drift?

### DISCUSSION THREAD

**Discussion by XaviMartiCamarasa** [online-ml/river#1076] [Q&A] [answered]
HI. I'm new at RiverML framework, so my questions could be naive, but... sorry for that!

How can I retrain a model after checking Concept Drift?

Many thanks in advance

**Comment by MaxHalford** [comment, marked as answer]
Hello @XaviMartiCamarasa, don't be sorry! We're all learning :)

This is a bit of 1 million dollar question. I'm not sure about your context, so it's difficult to answer in detail. But in general, you would probably want to re-train your model after the concept drift.

In River you can `clone` a model. By default it will copy the model with its input parameters, but not the model's state. See [here](https://riverml.xyz/0.14.0/recipes/cloning-and-mutating/) for more information.

**Reply by XaviMartiCamarasa** [reply]
It is just what I tried to ask. Many Thanks MaxHalford!!

**Reply by XaviMartiCamarasa** [reply]
So, having the model cloned because of a ConcepDrift..What the new training data should be?
How/Where can I find it?

Do I need to open another question for this? O We can keep wrinting on this?

Kind Regards

**Reply by MaxHalford** [reply]
> So, having the model cloned because of a ConcepDrift..What the new training data should be?
How/Where can I find it?

I'm not sure I understand what you mean. But if you have a stream of data, your training data is simply everything that arrives after the moment you detect a concept drift.

> Do I need to open another question for this? O We can keep wrinting on this?

It's fine keep going here.

**Reply by XaviMartiCamarasa** [reply]
Hi MaxHalford.
How do I know, the data after detecting de drift?

Regards

**Reply by MaxHalford** [reply]
I'm not sure I understand. You have a stream of timestamped data, right? Well the data after the drift is simply the data after the timestamp where the drift occurred. Doesn't that make sense?

**Reply by XaviMartiCamarasa** [reply]
Let me think about it :)

**Reply by XaviMartiCamarasa** [reply]
> I'm not sure I understand. You have a stream of timestamped data, right? Well the data after the drift is simply the data after the timestamp where the drift occurred. Doesn't that make sense?

It makes a lot of sense!
